axelrod/strategies/strategy_transformers.py

```diff
@@ -0,0 +1,202 @@
+                proposed_action = PlayerClass.strategy(opponent)
+            else:
+                proposed_action = PlayerClass.strategy(self, opponent)
+            # Apply the wrapper
+            return strategy_wrapper(self, opponent, proposed_action,
+                                    *wrapper_args, **wrapper_kwargs)
+
+        # Define a new class and wrap the strategy method
+        # Modify the PlayerClass name
+        new_class_name = PlayerClass.__name__
+        name = PlayerClass.name
+        if name_prefix:
+            # Modify the Player name (class variable inherited from Player)
+            new_class_name = name_prefix + PlayerClass.__name__
+            # Modify the Player name (class variable inherited from Player)
+            name = name_prefix + ' ' + PlayerClass.name
+        # Dynamically create the new class
+        new_class = type(new_class_name, (PlayerClass,),
+                         {"name": name, "strategy": strategy})
+        return new_class
+    return decorate
+
+def flip_wrapper(player, opponent, action):
+    """Applies flip_action at the class level."""
+    return flip_action(action)
+
+FlipTransformer = StrategyTransformerFactory(flip_wrapper, name_prefix="Flipped")
+
+def noisy_wrapper(player, opponent, action, noise=0.05):
+    """Applies flip_action at the class level."""
+    r = random.random()
+    if r < noise:
+        return flip_action(action)
+    return action
+
+def NoisyTransformer(noise):
+    return StrategyTransformerFactory(noisy_wrapper,
+                                      wrapper_args=(noise,),
+                                      name_prefix="Noisy")
+
+def forgiver_wrapper(player, opponent, action, p):
+    """If a strategy wants to defect, flip to cooperate with the given
+    probability."""
+    if action == D:
+        return random_choice(p)
+    return C
+
+def ForgiverTransformer(p):
+    return StrategyTransformerFactory(forgiver_wrapper, wrapper_args=(p,))
+
+def initial_sequence(player, opponent, action, initial_seq):
+    """Play the moves in `seq` first (must be a list), ignoring the strategy's
+    moves until the list is exhausted."""
+    index = len(player.history)
+    if index < len(initial_seq):
+        return initial_seq[index]
+    return action
+
+## Defection initially three times
+def InitialTransformer(seq=None):
+    if not seq:
+        seq = [D] * 3
+    transformer = StrategyTransformerFactory(initial_sequence, wrapper_args=(seq,),
+                                             name_prefix="Initial")
+    return transformer
+
+def final_sequence(player, opponent, action, seq):
+    """Play the moves in `seq` first, ignoring the strategy's
+    moves until the list is exhausted."""
+    length = player.tournament_attributes["length"]
+
+    if length < 0: # default is -1
+        return action
+
+    index = length - len(player.history)
+    if index <= len(seq):
+        return seq[-index]
+    return action
+
+# Defect on last N actions
```

Comment:
Good catch -- that's what it was initially but then I generalized to an arbitrary sequence.